### PR TITLE
Increase buffer size for webflux client

### PR DESCRIPTION
**Pull request by blindpirate**
See
https://stackoverflow.com/questions/59735951/databufferlimitexception-exceeded-limit-on-max-bytes-to-buffer-webflux-error

### CODE REVIEW

File: application/src/main/resources/application.properties [hunk 1]
```diff
@@ -5,3 +5,4 @@ teamcity.api.token=${TEAMCITY_API_TOKEN}
 spring.datasource.url=${JDBC_DATABASE_URL:jdbc:postgresql://localhost:5432/btdevprod}
 spring.datasource.username=${JDBC_DATABASE_USERNAME:btdevprod}
 spring.datasource.password=${JDBC_DATABASE_PASSWORD}
+spring.codec.max-in-memory-size=10MB
```
Comment (wolfs) [inline application/src/main/resources/application.properties:8]: Can we do this only in the Teamcity client? The GE export API client isn't affect, so I'd like to continue there with the small buffer size.


I also find 10MB pretty large, given that the default is 256kb. How about we go for 512kb first?

Comment (blindpirate) [reply application/src/main/resources/application.properties]: Better now?

File: teamcity-export/src/main/kotlin/org/gradle/devprod/collector/teamcity/TeamcityClientService.kt [hunk 2]
```diff
@@ -29,7 +30,12 @@ class TeamcityClientService(
 
     private val teamCityRestApiBuildsUrl = "https://builds.gradle.org/app/rest/builds"
 
-    private val client: WebClient = WebClient.create(teamCityRestApiBuildsUrl)
+    private val client: WebClient = WebClient.builder().baseUrl(teamCityRestApiBuildsUrl)
+        .exchangeStrategies(
+            ExchangeStrategies.builder()
+                .codecs { it.defaultCodecs().maxInMemorySize(512 * 1024) }
+                .build()
+        ).build();
```
Comment (wolfs) [inline teamcity-export/src/main/kotlin/org/gradle/devprod/collector/teamcity/TeamcityClientService.kt:38]:
```suggestion
        .codecs {  configurer -> configurer.defaultCodecs().maxInMemorySize(512 * 1024) }
        .build();
```

How about this?

File: teamcity-export/src/main/kotlin/org/gradle/devprod/collector/teamcity/TeamcityClientService.kt [hunk 3]
```diff
@@ -29,7 +30,12 @@ class TeamcityClientService(
 
     private val teamCityRestApiBuildsUrl = "https://builds.gradle.org/app/rest/builds"
 
-    private val client: WebClient = WebClient.create(teamCityRestApiBuildsUrl)
+    private val client: WebClient = WebClient.builder().baseUrl(teamCityRestApiBuildsUrl)
```
Comment (wolfs) [inline teamcity-export/src/main/kotlin/org/gradle/devprod/collector/teamcity/TeamcityClientService.kt:33]: I wonder if the Enterprise client needs this or the teamcity client. When trying locally, it seemed like the Enterprise client need it.

File: teamcity-export/src/main/kotlin/org/gradle/devprod/collector/teamcity/TeamcityClientService.kt (hunk 2 above)
Comment (blindpirate) [reply teamcity-export/src/main/kotlin/org/gradle/devprod/collector/teamcity/TeamcityClientService.kt]: Done.

File: teamcity-export/src/main/kotlin/org/gradle/devprod/collector/teamcity/TeamcityClientService.kt (hunk 3 above)
Comment (blindpirate) [reply teamcity-export/src/main/kotlin/org/gradle/devprod/collector/teamcity/TeamcityClientService.kt]: Yeah... looking at the failure it seems to come from GE client.

**Review by wolfs — COMMENTED** [review]
LGTM!

**Pull request merged**